airflow/providers/amazon/provider.yaml

```diff
@@ -710,3 +710,16 @@ additional-extras:
   - name: cncf.kubernetes
     dependencies:
       - apache-airflow-providers-cncf-kubernetes>=7.2.0
+
+config:
+  aws:
+    description: This section applies settings for Amazon Web Services (AWS) integration.
+    options:
+      session_factory:
+        description: |
+          Full import path to the class which implements custom session factory for ``boto3.session.Session``,
+          for more detail please have a look at :ref:`howto/connection:aws:session-factory`.
+        default: ~
+        example: my_company.aws.MyCustomSessionFactory
+        type: string
+        version_added: ~
```

Comment:
If that is the case, should we add a check to make sure this value is not higher than the latest airflow version?

Reply:
I guess it is additional validation just in case for Airflow Core
E.g.: option would added to 2.8, but in 2.7.x it do not make any affect on configuration side .

https://github.com/apache/airflow/blob/b82ce61285f3f4f0c7eccb2c3effaef53c9fb84e/airflow/configuration.py#L571-L576

And this remove futures configs, during generation documentatio 
https://github.com/apache/airflow/blob/2663b177a666ee150f86bd16ad51526319fa7adb/docs/conf.py#L434-L442

In general this logic designed in the time, when providers doesn't have their own documentation for configuration.